### PR TITLE
Improves Search and Show Number Find

### PR DESCRIPTION
**Pull request by stolinski**
Fixes all of #1668

### CODE REVIEW

**Comment by stolinski** [comment]
I have no idea why the pw tests are suddenly failing here. It looks like some kind of out of memory issue? I ran them locally and they all passed. Nohting I changed will affect that, so maybe we are just good to merge and figure out why they aren't passing after?

File: src/routes/(site)/[...number]/+page.server.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { prisma_client } from '$/server/prisma-client';
+import get_show_path from '$/utilities/slug';
+import { error, redirect } from '@sveltejs/kit';
+
+export async function load(event) {
```
Comment (w3cj) [inline src/routes/(site)/[...number]/+page.server.ts:5]: Extract this into `lib` so it can be re-used in both spots without copy / paste.

**Review by w3cj — CHANGES_REQUESTED** [review]

**Comment by wesbos** [comment]
LG™ - I tested that the redirects still work

**Review by wesbos — APPROVED** [review]

File: src/routes/(site)/[...number]/+page.server.ts (hunk 1 above)
Comment (stolinski) [reply src/routes/(site)/[...number]/+page.server.ts:5]: Going to mark as resolved, but please re-open if you aren't cool with that.

**Pull request merged**